packages/react-instantsearch/src/connectors/connectCurrentRefinements.js

```diff
@@ -8,20 +8,22 @@ import createConnector from '../core/createConnector';
  * @kind connector
  * @category connector
  * @providedPropType {function} refine - a function to remove a single filter
- * @providedPropType {array.<{key: string, label: string}>} filters - all the filters, the key for calling the refine prop function, label is for the display.
+ * @providedPropType {array.<{label: string, attributeName: string, currentRefinement: string || object, items: array, value: function}>} items - all the filters, the `value` is to pass to the `refine` function for removing all currentrefinements, `label` is for the display. When existing several refinements for the same atribute name, then you get a nested `items` object that contains a `label` and a `value` function to use to remove a single filter. `attributeName` and `currentRefinement` are metadata containing row values.
  */
 export default createConnector({
   displayName: 'AlgoliaCurrentRefinements',
 
   getProps(props, state, search, metadata) {
     return {
       items: metadata.reduce((res, meta) =>
-        typeof meta.filters !== 'undefined' ? res.concat(meta.filters) : res
-      , []),
+          typeof meta.items !== 'undefined' ? res.concat(meta.items) : res
+        , []),
     };
   },
 
-  refine(props, state, filters) {
-    return filters.reduce((res, filter) => filter.clear(res), state);
+  refine(props, state, items) {
+    // `value` corresponds to our internal clear function computed in each connector metadata.
+    const clears = items instanceof Array ? items.map(item => item.value) : [items];
```

Comment:
refine here has one purposes: calling the clear function to remove refinements. 
We hide this to the end user by passing a clear function as value, then he need to use it like this: props.refine(item.value).

However we have two use cases: either we directly have the clear function if it's a single clear, or we get an array of items, and then the idea is to call each clear function to remove all filters at the same time (clearAll behavior).

I'm just unhappy with the naming I found to express this in the code.